File: app/src/main/java/com/prongbang/bottomsheetdialog/activities/MainActivity.java
package com.prongbang.bottomsheetdialog.activities;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.prongbang.bottomsheetdialog.R;
import com.prongbang.bottomsheetdialog.widget.MyBottomSheetDialog;

public class MainActivity extends BaseActivity implements View.OnClickListener {

    private TextView tvClickMe;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvClickMe = (TextView) findViewById(R.id.tvClickMe);
        tvClickMe.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tvClickMe:

                MyBottomSheetDialog myBottomSheetDialog = MyBottomSheetDialog.getInstance(this);
                myBottomSheetDialog.setTvTitle("Bottom Sheet Dialog");
                myBottomSheetDialog.setTvSubTitle("Read more...");
                myBottomSheetDialog.setIvAvatar("https://avatars3.githubusercontent.com/u/6635954?v=3&u=d18aab686938ecda4b96f29e4e3b776008ced91f&s=400");
                myBottomSheetDialog.setCanceledOnTouchOutside(false);
                myBottomSheetDialog.show();

                break;
        }
    }
}

File: app/src/main/java/com/prongbang/bottomsheetdialog/MainActivity.kt
package com.prongbang.bottomsheetdialog

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

	private val avatarUrl = "https://avatars3.githubusercontent.com/u/6635954?v=3&u=d18aab686938ecda4b96f29e4e3b776008ced91f&s=400"

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)

		tvClickMe.setOnClickListener {
			showBottomSheetDialog()
		}
	}

	private fun showBottomSheetDialog() {
		MyBottomSheetDialogFragment.Builder(supportFragmentManager)
				.setAvatar(avatarUrl)
				.setTitle("เดฟไปวันๆ")
				.setSubtitle("https://prongbang.github.io")
				.setOnCloseButton {
					Toast.makeText(this, "Bottom Sheet Dialog Closed", Toast.LENGTH_SHORT).show()
				}
				.build()
				.show()
	}

}